plugin/adsk/scripts/mayaUsdMenu.mel

```diff
@@ -154,6 +241,116 @@ global proc mayaUsdMenu_createMenuCallback() {
+    string $selectionKind = mayaUSD_getSelectionKind();
+    switch($selectionKind) {
+        case "":
+            menuItem -e -radioButton on selectKindNone;
+            break;
+        case "model":
+            menuItem -e -radioButton on selectKindModel;
+            break;
+        case "group":
+            menuItem -e -radioButton on selectKindGroup;
+            break;
+        case "assembly":
+            menuItem -e -radioButton on selectKindAssembly;
+            break;
+        case "component":
+            menuItem -e -radioButton on selectKindComponent;
+            break;
+        case "subcomponent":
+            menuItem -e -radioButton on selectKindSubcomponent;
+            break;
+        default:
+            // check for custom kind, the menuItem's label
+            // must match the kind
+            global string $gMayaUsdSelectModeSubMenu;
+            $menuItems = `menu -q -itemArray $gMayaUsdSelectModeSubMenu`;
+            for($i = 0; $i < size($menuItems); $i++) {
+                $label = `menuItem -q -label $menuItems[$i]`;
+                if($label == $selectionKind) {
+                    menuItem -e -radioButton on $menuItems[$i];
+                    break;
+                }
+            }
+    }
+}
+
+///////////////////////////////////////////////////////////////////////////////
+// mayaUsdMenu_selectMenuCallback
+// setup the items in Maya's "Select" menu
+global proc mayaUsdMenu_selectMenuCallback() {
+
+    global string $gMayaUsdSelectMenuDivider;
+    if(!`menuItem -q -exists $gMayaUsdSelectMenuDivider`) {
+        string $label = `getMayaUsdString("kUniversalSceneDescription")`;
+        $gMayaUsdSelectMenuDivider = `menuItem -divider true -dividerLabel $label`;
+    }
+
+    global string $gMayaUsdSelectModeSubMenu;
+    if(!`menuItem -q -exists $gMayaUsdSelectModeSubMenu`) {
+        string $mayaVersion = getMayaMajorVersion();
+      
+        string $label = `getMayaUsdString("kUSDSelectionMode")`;
+        string $ann = `getMayaUsdString("kUSDSelectionModeAnn")`;
+        $gMayaUsdSelectModeSubMenu = `menuItem -label $label -annotation $ann -subMenu true -tearOff true -version $mayaVersion`;
+
+        $label = `getMayaUsdString("kKind")`;
+        menuItem -divider true -dividerLabel $label;
+        radioMenuItemCollection;
+
+        // add the default (built-in) select kind menu items
+        menuItem -radioButton on  -runTimeCommand ("mayaUsdSelectKindNone") selectKindNone;
+        menuItem -radioButton off -runTimeCommand ("mayaUsdSelectKindModel") selectKindModel;
+        menuItem -radioButton off -runTimeCommand ("mayaUsdSelectKindGroup") selectKindGroup;
+        menuItem -radioButton off -runTimeCommand ("mayaUsdSelectKindAssembly") selectKindAssembly;
+        menuItem -radioButton off -runTimeCommand ("mayaUsdSelectKindComponent") selectKindComponent;
+        menuItem -radioButton off -runTimeCommand ("mayaUsdSelectKindSubComponent") selectKindSubcomponent;
```

Comment:
What is the purpose of creating/having all these runtime commands? Why not just call the MEL proc mayaUSD_setSelectionKind with string just like you do below on line 330 for the custom kinds?

Reply:
They are built-in kind ... so many user will want to bind a shortcut  key ? I think in maya to be able to bind a shortcut key we need to use runtime command

Reply:
To bind to a hotkey you need a **nameCommand** and for that you can set any command, but normally in Maya we use runtimeCommand for these. I see you are following the pattern of other menu items in this file so I'm good with it.